code/game/objects/item.dm

```diff
@@ -806,9 +806,27 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 	update_icon()
 	update_clothing_icon()
 
+// Used to call appropriate slot updates in update_clothing_icon()
+/obj/item/proc/get_associated_equipment_slots()
+	SHOULD_CALL_PARENT(TRUE)
+	if(item_flags & ITEM_FLAG_IS_BELT)
+		LAZYADD(., slot_belt_str)
+
 // Updates the icons of the mob wearing the clothing item, if any.
 /obj/item/proc/update_clothing_icon()
-	return
+	var/equip_slots = get_associated_equipment_slots()
+	if(!equip_slots)
+		return FALSE
+	var/mob/wearer = loc
+	if(!istype(wearer))
+		return FALSE
+	if(islist(equip_slots))
+		for(var/slot in equip_slots)
+			wearer.update_equipment_overlay(slot, FALSE)
+		wearer.update_icon()
+	else
```

Comment:
As far as I know, this `else` case can never be triggered now. Either `get_associated_equipment_slots` returns null or a list. Null is checked earlier.